sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/SwitchDefaultClauseEmpty.cs

```diff
@@ -55,14 +55,22 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                         return;
                     }
 
-                    var breakStatement = section.Statements.First();
-
-                    if (breakStatement.IsKind(SyntaxKind.BreakStatement))
+                    if (section.Statements[0].IsKind(SyntaxKind.BreakStatement) &&
+                        !HasAnyComment(section))
                     {
                         c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, section.GetLocation()));
                     }
                 },
                 SyntaxKind.SwitchSection);
         }
+
+        private static bool HasAnyComment(SwitchSectionSyntax section) =>
+            section.Labels.Last()
+                .GetTrailingTrivia()                                // handle comments after default:

```

Comment:
Surprisingly, _default_ doesn't have to be the last label in a switch section.
What do we want the rule to do if there are multiple labels in the section containing _default_? Is it enough that there is a comment on any of the labels, or before after the _break_ statement? (I think probably yes).

Reply:
I know this doesn't have to be but then we go into really complex corner cases.
```
case A:
default:
case B:
    // Some comment
```

Does the comment needs to be besides `default`? Which seems weird because there is another case right after... 

Also as we already filtered out the default case we know that when this method is called we DO have the default case in it so I have assumed that the user would probably put the comment in front of the last case.

Reply:
Fair enough, but it might be better to update the comment to call the assumption out e.g. _// handle comments after the last label, which will normally be default_

Reply:
ok